openttd/Dockerfile

```diff
@@ -1,19 +1,31 @@
 FROM base
 
-RUN apt-get install -y libsdl1.2debian
-RUN apt-get install -y libfontconfig1
+RUN apt-get update && \
+	apt-get install -y libsdl1.2debian \
+	libfontconfig1 && \
+	adduser openttd && \
+	mkdir -p /home/openttd/extract && \
+	chown -R openttd:openttd /home/openttd/extract
 
-RUN adduser openttd && mkdir -p /openttd/extract && chown -R openttd:openttd /openttd
-WORKDIR /openttd
+USER openttd
+WORKDIR /home/openttd
 
-ADD opengfx*.zip openmsx*.zip opensfx*.zip openttd*.tar.gz /openttd/
-RUN mv openttd-*-linux-generic-amd64/* .
-RUN unzip openmsx*.zip -d /openttd/extract/
-RUN unzip opensfx*.zip -d /openttd/extract/
-RUN unzip opengfx*.zip -d /openttd/extract/ && tar -xf /openttd/extract/opengfx*.tar -C /openttd/extract/
-RUN mv --no-clobber /openttd/extract/*/* /openttd/baseset/
+ADD download.sh .
 
-RUN chown -R openttd:openttd /openttd
-USER openttd
+RUN ./download.sh && \
```

Comment:
It doesn't have to redownload _every_ time, no, but it does have to redownload every time I update base or build on a different server or do dev work on the Dockerfile, or otherwise invalidate that cache, and with the size of some of the assets used in this repo, that's a problem.

It's not The Docker Way, but it's an acceptable pragmatic compromise between The Right Way and usability/flexibility, so I use the pre-build download pattern as much as possible.

No problems on the time thing - there's no urgency at all. And I can go through and make the changes I requested myself if you don't have time, no problem :)